pkg/tools/cli/loadtest/service/loader.go

```diff
@@ -40,20 +42,21 @@ type Loader interface {
 }
 
 type (
-	requestFunc func(assets.Dataset) ([]interface{}, error)
-	loaderFunc  func(context.Context, vald.ValdClient, interface{}, ...grpc.CallOption) error
+	loadFunc func(context.Context, *grpc.ClientConn, interface{}, ...grpc.CallOption) (interface{}, error)
 )
 
 type loader struct {
```

Comment:
I think `loader`'s responsibility is wide.
so I think it should be split.
for example, loader should have only `Do` and `Prepare `.
and switch the method of getting loadFunc and dataProvider via the interface.

```
pkg/tools/cli/loadtest/service/loader.go
type Provider interface {
    LoderFunc() ....
    DataFunc() ....
}

pkg/tools/cli/loadtest/service/provider/provider.go

type provider struct {
    cfg *config.Config
}

fund New(opts ...Option) service.Provider {
....
}


```

Reply:
This sounds good.
However, when fixing it in this PR, I think it is too large.
How about it?

Reply:
ok, It seems to be good in the direction of responding with another PR.

cc: @kpango